cmd/soroban-rpc/internal/methods/simulate_transaction.go

```diff
@@ -35,11 +35,11 @@ type RestorePreamble struct {
 
 type SimulateTransactionResponse struct {
 	Error           string                       `json:"error,omitempty"`
-	TransactionData string                       `json:"transactionData"` // SorobanTransactionData XDR in base64
-	MinResourceFee  int64                        `json:"minResourceFee,string"`
+	TransactionData string                       `json:"transactionData,omitempty"` // SorobanTransactionData XDR in base64
+	MinResourceFee  int64                        `json:"minResourceFee,string,omitempty"`
 	Events          []string                     `json:"events,omitempty"`          // DiagnosticEvent XDR in base64
 	Results         []SimulateHostFunctionResult `json:"results,omitempty"`         // an array of the individual host function call results
-	Cost            SimulateTransactionCost      `json:"cost"`                      // the effective cpu and memory cost of the invoked transaction execution.
+	Cost            SimulateTransactionCost      `json:"cost,omitempty"`            // the effective cpu and memory cost of the invoked transaction execution.
```

Comment:
might be missing something here: won't this always be present since it's a struct? (and same for `RestorePreamble`, below, actually)

Reply:
unless the operation execution was successful, there wouldn't be any `CPUInstructions` or `MemoryBytes`, wouldn't it ?

Reply:
If there is an error there is no point in sending cost.